File: retrofit/src/main/java/webconnect/com/webconnect/WebParam.java
package webconnect.com.webconnect;

import android.app.Activity;
import android.app.Dialog;
import android.content.Context;

import java.io.File;
import java.io.Serializable;
import java.util.LinkedHashMap;
import java.util.Map;


/**
 * The type Web param.
 */
public class WebParam implements Serializable {
    /**
     * The Activity context.
     */
    Activity activityContext;
    /**
     * The Context.
     */
    Context context;
    /**
     * The Url.
     */
    String url, /**
     * The Base url.
     */
    baseUrl;
    /**
     * The Http type.
     */
    HttpType httpType = HttpType.GET;
    /**
     * The Request param.
     */
    Map<String, ?> requestParam = new LinkedHashMap<>();

    /**
     * The Multipart param.
     */
    Map<String, String> multipartParam = new LinkedHashMap<>();
    Map<String, File> multipartParamFile = new LinkedHashMap<>();
    /**
     * The Header param.
     */
    Map<String, String> headerParam = new LinkedHashMap<>();
    /**
     * The Callback.
     */
    WebHandler.OnWebCallback callback;
    /**
     * Progress during download
     */
    WebHandler.ProgressListener progressListener;

    WebHandler.AnalyticsListener analyticsListener;
    /**
     * Progress Dialog
     */
    Dialog dialog;
    /**
     * The Model.
     */
    Class<?> model;
    /**
     * The Error.
     */
    Class<?> error;
    /**
     * The Task id.
     */
    int taskId;

    long connectTimeOut = 0, readTimeOut = 0;
    /**
     * Download File anyType
     */
    boolean isCacheEnabled = false;
    File file;

    /**
     * The enum Http type.
     */
    public enum HttpType {
        /**
         * Get http type.
         */
        GET, /**
         * Post http type.
         */
        POST, /**
         * Put http type.
         */
        PUT,
        /**
         * Patch http type.
         */
        PATCH,
        /**
         * Delete http type.
         */
        DELETE,
        /**
         * head http type.
         */
        HEAD,
        /**
         * Options http type.
         */
        OPTIONS,
        /**
         * Multipart type
         */
        MULTIPART,
        /**
         * Download
         */
        DOWNLOAD


    }


    public Activity getActivityContext() {
        return activityContext;
    }

    public Context getContext() {
        return context;
    }

    public String getUrl() {
        return url;
    }

    public String getBaseUrl() {
        return baseUrl;
    }

    public HttpType getHttpType() {
        return httpType;
    }

    public Map<String, ?> getRequestParam() {
        return requestParam;
    }

    public Map<String, String> getMultipartParam() {
        return multipartParam;
    }

    public Map<String, File> getMultipartParamFile() {
        return multipartParamFile;
    }

    public Map<String, String> getHeaderParam() {
        return headerParam;
    }

    public WebHandler.OnWebCallback getCallback() {
        return callback;
    }

    public Class<?> getModel() {
        return model;
    }

    public Class<?> getError() {
        return error;
    }

    public int getTaskId() {
        return taskId;
    }

    public File getFile() {
        return file;
    }

    public WebHandler.ProgressListener getProgressListener() {
        return progressListener;
    }

    public Dialog getDialog() {
        return dialog;
    }

    public boolean isCacheEnabled() {
        return isCacheEnabled;
    }

    public WebHandler.AnalyticsListener getAnalyticsListener() {
        return analyticsListener;
    }
}

File: retrofit/src/main/java/webconnect/com/webconnect/WebParam.kt
package webconnect.com.webconnect

import android.app.Activity
import android.app.Dialog
import android.content.Context
import java.io.File
import java.io.Serializable
import java.util.*


/**
 * The type Web param.
 */
class WebParam : Serializable {
    /**
     * The Activity context.
     */
    var activityContext: Activity? = null
        internal set
    /**
     * The Context.
     */
    var context: Context? = null
        internal set
    /**
     * The Url.
     */
    var url: String? = null
        internal set
    /**
     * The Base url.
     */
    var baseUrl: String? = null
        internal set
    /**
     * The Http type.
     */
    var httpType = HttpType.GET
        internal set
    /**
     * The Request param.
     */
    var requestParam: Map<String, Any> = LinkedHashMap<String, Any>()
        internal set

    /**
     * The Multipart param.
     */
    var multipartParam: Map<String, String> = LinkedHashMap()
        internal set
    var multipartParamFile: Map<String, File> = LinkedHashMap()
        internal set
    /**
     * The Header param.
     */
    var headerParam: Map<String, String> = LinkedHashMap()
        internal set
    /**
     * The Callback.
     */
    var callback: WebHandler.OnWebCallback? = null
        internal set
    /**
     * Progress during download
     */
    var progressListener: WebHandler.ProgressListener? = null
        internal set

    var analyticsListener: WebHandler.AnalyticsListener? = null
        internal set
    /**
     * Progress Dialog
     */
    var dialog: Dialog? = null
        internal set
    /**
     * The Model.
     */
    var model: Class<*> = Object::class.java
        internal set
    /**
     * The Error.
     */
    var error: Class<*> = Object::class.java
        internal set
    /**
     * The Task id.
     */
    var taskId: Int = 0
        internal set

    var connectTimeOut: Long = 0
        internal set
    var readTimeOut: Long = 0
        internal set
    /**
     * Download File anyType
     */
    var isCacheEnabled = false
        internal set
    var file: File? = null
        internal set

    /**
     * The enum Http type.
     */
    enum class HttpType {
        /**
         * Get http type.
         */
        GET,
        /**
         * Post http type.
         */
        POST,
        /**
         * Put http type.
         */
        PUT,
        /**
         * Patch http type.
         */
        PATCH,
        /**
         * Delete http type.
         */
        DELETE,
        /**
         * head http type.
         */
        HEAD,
        /**
         * Options http type.
         */
        OPTIONS,
        /**
         * Multipart type
         */
        MULTIPART,
        /**
         * Download
         */
        DOWNLOAD


    }
}
